train/ngram.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+public class Ngram
+{
+    public void ngram_n2(string sentence, int N)
+    {
+        string[] Sentence = sentence.Split(" ");
+        //split to single word
+        List<string> completed = new List<string>();
+        //save the result
+        if (sentence == null)
+        {
+            Console.WriteLine("The test sentence is empty");
+        }
+        else
+        {
+            for (int i = 0; i <= (Sentence.Length - N); i++)
+            //loop word list
+            {
+                List<string> temp = new List<string>();
+                for (int j = i; j < i + N; j++)
+                //complete one n-gram each loop
+                {
+                    temp.Add(Sentence[j]);
+                }
+                completed.Add(string.Join(" ", temp));
+            }
+            for (int i = 0; i < completed.Count; i++)
+            {
+                Console.WriteLine(completed[i]);
+            }
+        }
+    }
+    public IEnumerable<string> ngram_queue(string sentence, int N)
+    {
+        sentence = sentence + " ";
+        StringBuilder completed = new StringBuilder();
+        //save the result
+        Queue<int> Word_len = new Queue<int>();
+        //save the length of each word
+        int count = 0;
+        //count the word have been catched
+        int lastWord = 0;
+        //the length of lastword in queue
+        if (sentence == null)
```

Comment:
I'm a little confused here. Why not just check if the sentence is null or empty (or full of white spaces) at the start of the method?